src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php

```diff
@@ -81,20 +81,26 @@ public function setFile($file)
      */
     public function pathToLogFile($file)
     {
-        $logsPath = $this->storage_path;
-        $logsPath .= ($this->folder) ? '/' . $this->folder : '';
-
         if (app('files')->exists($file)) { // try the absolute path
             return $file;
         }
-
-        $file = $logsPath . '/' . $file;
-
-        // check if requested file is really in the logs directory
-        if (dirname($file) !== $logsPath) {
-            throw new \Exception('No such log file');
+        if (is_array($this->storage_path)) {
+            foreach ($this->storage_path as $value) {
+                if (app('files')->exists($value . '/' . $file)) { // try the absolute path
+                    $file = $value . '/' . $file;
+                    break;
+                }
+            }
+        }
+        if (!is_array($this->storage_path)) {
```

Comment:
Why not if/else?

Reply:
Codacy recommends "The method getFolders uses an else expression. Else is never necessary and you can simplify the code to work without else."
I just put it back (Still checking the comment above, I will write some test cases for that)

Reply:
Yes you are right about pointing this:

> The method getFolders uses an else expression. Else is never necessary and you can simplify the code to work without else.

Still, it does not mean it should b replaced by another `if`. They says that it could be simplified. The usual example for this is replacing:

```
if ($a) {
    return 'a';
} else {
    return 'b';
}
```

With: 

```
if ($a) {
    return 'a';
}
return 'b';
```

This simplifies the code. Replacing `else` with `if (!thing_i_tested_before())` hacks the Codacy detector but does not simplify code I think (but maybe I'm wrong)